src/components/search-input/search-input.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import "./search-input.scss";
+import { FaSearch } from "react-icons/fa";
+
+export const SearchInput = ({
+  hasIcon = true,
+  onValueChange,
+  placeholder = "cauta",
+  value
+}) => {
+  const [currentValue, setCurrentValue] = useState(0);
+  return (
+    <div className={`__search-input-container ${hasIcon ? "__has_icon" : ""}`}>
+      <input
+        placeholder={placeholder}
+        value={value || currentValue || ""}
```

Comment:
Hmm, this can be very error prone. We should not rely on the state from outside the component. If we want to have a predefined value for the input, the way to go is when we initialize the component, we check once if we have `value`, then update the `currentValue`.

```js
  useEffect(() => {
    if (value) {
      setCurrentValue(value);
    }
  }, []);
```
Then the `onChange` will become
```js
  onChange={event => {
    if (onValueChange) {
      onValueChange(event.target.value);
    }

    setCurrentValue(event.target.value);
  }}
```

Please let me know if this makes sense